deployed_model/app.py

```diff
@@ -58,82 +57,89 @@ def unpickle(model_file : str):
-            
-            if(n < 6):
-                abilities.append(contents.title())
-            elif(n==6):
+            if(n==0):
                 hp = int(contents)
-            elif(n==7):
+            elif(n==1):
+                attack = int(contents)
+            elif(n==2):
+                defense = int(contents)
+            elif(n==3):
+                sp_attack = int(contents)
+            elif(n==4):
+                sp_defense = int(contents)
+            elif(n==5):
                 speed = int(contents)
-            elif(n==8):
-                #print()
+            elif(n==6):
                 base_egg_steps = int(contents)
-            elif(n==9):
+            elif(n==7):
                 capture_rate = int(contents)
+            # elif(n==8):
+            #     xp = int(contents)
+            # elif(n==9):
+            #     weight = float(contents)
+            # elif(n==10):
+            #     height = float(contents)
+            # elif(n==11):
+            #     base_happiness = int(contents)
             n += 1
 
+        
+        
+        # create a dataframe and get it in the expected format
 
-        #print(abilities, hp, speed, base_egg_steps, capture_rate)
-        #print(abilities)
-    
-
-        #abilities = ['Pressure', 'Pickpocket']
-        # base_egg_steps = 250
-        # capture_rate = 50
-        # speed = 100
-        # hp = 50
+        base_total = hp + attack + defense + sp_attack + sp_defense + speed
+        #print(base_total, base_egg_steps, capture_rate)
 
         test_data = {
-            # 'abilities' : [['Pressure', 'Pickpocket']],
-            # 'base_egg_steps' : [250],
-            # 'capture_rate' : [50],
-            # 'hp' : [50],
-            # 'speed' : [100],
-            'abilities' : [abilities],
             'base_egg_steps' : [base_egg_steps],
             'capture_rate' : [capture_rate],
-            'hp' : [hp],
-            'speed' : [speed],
+            'base_total' : [base_total],
+            # 'base_happiness' : [base_happiness],
+            # 'height_m': [height],
+            # 'weight_kg' : [weight],
+            # 'experience_growth' : [xp],
         }
 
-        # create a dataframe and get it in the expected format
         test_df = pd.DataFrame(test_data)
-        mlb = MultiLabelBinarizer()
-        abilities_data = pd.DataFrame(  mlb.fit_transform(test_df['abilities']), columns=mlb.classes_, index=test_df.index)
-        test_df = test_df.drop(columns=['abilities'], axis=1)
-        test_df = pd.concat([test_df, abilities_data], axis=1)
-        test_df = test_df.reindex(columns=xs.columns, fill_value=0)
         return test_df
     except Exception as e:
         print(f"Input error: {e}")
-        return None
+        error_df = pd.DataFrame()
+        return error_df
 
 # Given a pandas dataframe in the a format used for training, predict if the pokemon is legendary
 
 def predict_is_legendary(df, model) -> bool:
     result = model.predict(df)
```

Comment:
I get this error every time i press predict:

Traceback (most recent call last):
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 305, in <module>
    main()
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 302, in main
    run_app(model, xs)
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 267, in run_app
    is_legendary = predict_is_legendary(test_df, model)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 141, in predict_is_legendary
    result = model.predict(df)
             ^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\sklearn\pipeline.py", line 600, in predict       
    Xt = transform.transform(Xt)
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\sklearn\utils\_set_output.py", line 316, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 26, in transform
    return xs[self.columns]
           ~~^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\frame.py", line 4108, in __getitem__ 
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\indexes\base.py", line 6200, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\indexes\base.py", line 6252, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['against_bug', 'against_dark', 'against_dragon', 'against_electric', 'against_fairy', 'against_fight', 'against_fire', 'against_flying', 'against_ghost', 'against_grass', 'against_ground', 'against_ice', 'against_normal', 'against_poison', 'against_psychic', 'against_rock', 'against_steel', 'against_water', 'attack', 'base_happiness', 'defense', 'experience_growth', 'height_m', 'hp', 'percentage_male', 'pokedex_number', 'sp_attack', 'sp_defense', 'speed', 'weight_kg', 'type1_dark', 'type1_dragon', 'type1_electric', 'type1_fairy', 'type1_fighting', 'type1_fire', 'type1_flying', 'type1_ghost', 'type1_grass', 'type1_ground', 'type1_ice', 'type1_normal', 'type1_poison', 'type1_psychic', 'type1_rock', 'type1_steel', 'type1_water', 'type2_bug', 'type2_dark', 'type2_dragon', 'type2_electric', 'type2_fairy', 'type2_fighting', 'type2_fire', 'type2_flying', 'type2_ghost', 'type2_grass', 'type2_ground', 'type2_ice', 'type2_normal', 'type2_poison', 'type2_psychic', 'type2_rock', 'type2_steel', 'type2_water', 'Adaptability', 'Aftermath', 'Air Lock', 'Analytic', 'Anger Point', 'Anticipation', 'Arena Trap', 'Aroma Veil', 'Bad Dreams', 'Battery', 'Battle Armor', 'Battle Bond', 'Beast Boost', 'Berserk', 'Big Pecks', 'Blaze', 'Bulletproof', 'Cheek Pouch', 'Chlorophyll', 'Clear Body', 'Cloud Nine', 'Color Change', 'Comatose', 'Competitive', 'Compoundeyes', 'Contrary', 'Corrosion', 'Cursed Body', 'Cute Charm', 'Damp', 'Dancer', 'Dark Aura', 'Dazzling', 'Defeatist', 'Defiant', 'Download', 'Drizzle', 'Drought', 'Dry Skin', 'Early Bird', 'Effect Spore', 'Emergency Exit', 'Fairy Aura', 'Filter', 'Flame Body', 'Flare Boost', 'Flash Fire', 'Flower Gift', 'Flower Veil', 'Fluffy', 'Forewarn', 'Friend Guard', 'Frisk', 'Full Metal Body', 'Fur Coat', 'Gale Wings', 'Galvanize', 'Gluttony', 'Gooey', 'Grass Pelt', 'Grassy Surge', 'Guts', 'Harvest', 'Healer', 'Heatproof', 'Heavy Metal', 'Honey Gather', 'Huge Power', 'Hustle', 'Hydration', 'Hyper Cutter', 'Ice Body', 'Illuminate', 'Illusion', 'Immunity', 'Imposter', 'Infiltrator', 'Innards Out', 'Inner Focus', 'Insomnia', 'Intimidate', 'Iron Barbs', 'Iron Fist', 'Justified', 'Keen Eye', 'Klutz', 'Leaf Guard', 'Levitate', 'Light Metal', 'Lightningrod', 'Limber', 'Liquid Ooze', 'Liquid Voice', 'Long Reach', 'Magic Bounce', 'Magic Guard', 'Magician', 'Magma Armor', 'Magnet Pull', 'Marvel Scale', 'Mega Launcher', 'Merciless', 'Minus', 'Misty Surge', 'Mold Breaker', 'Moody', 'Motor Drive', 'Moxie', 'Multiscale', 'Mummy', 'Natural Cure', 'No Guard', 'Normalize', 'Oblivious', 'Overcoat', 'Overgrow', 'Own Tempo', 'Pickpocket', 'Pickup', 'Pixilate', 'Plus', 
'Poison Heal', 'Poison Point', 'Poison Touch', 'Power of Alchemy', 'Prankster', 'Pressure', 'Prism Armor', 'Protean', 'Psychic Surge', 'Pure Power', 'Queenly Majesty', 'Quick Feet', 'RKS System', 'Rain Dish', 'Rattled', 'Receiver', 'Reckless', 'Refrigerate', 'Regenerator', 'Rivalry', 'Rock Head', 'Rough Skin', 'Run Away', 'Sand Force', 'Sand Rush', 'Sand Stream', 'Sand Veil', 'Sap Sipper', 'Schooling', 'Scrappy', 'Serene Grace', 'Shadow Shield', 'Shadow Tag', 'Shed Skin', 'Sheer Force', 'Shell Armor', 'Shield Dust', 'Shields Down', 'Simple', 'Skill Link', 'Slow Start', 'Slush Rush', 'Sniper', 'Snow Cloak', 'Snow Warning', 'Solar Power', 'Solid Rock', 'Soul-Heart', 'Soundproof', 'Speed Boost', 'Stakeout', 'Stall', 'Stamina', 'Stance Change', 'Static', 'Steadfast', 'Steelworker', 'Stench', 'Sticky Hold', 'Storm Drain', 'Strong Jaw', 'Sturdy', 'Suction Cups', 'Super Luck', 'Surge Surfer', 'Swarm', 'Sweet Veil', 'Swift Swim', 'Symbiosis', 'Synchronize', 'Tangled Feet', 'Tangling Hair', 'Technician', 'Telepathy', 'Teravolt', 'Thick Fat', 'Tinted Lens', 'Torrent', 'Tough Claws', 'Toxic Boost', 'Trace', 'Triage', 'Truant', 'Turboblaze', 'Unaware', 'Unburden', 'Unnerve', 'Victory Star', 'Vital Spirit', 'Volt Absorb', 'Water Absorb', 'Water Bubble', 'Water Compaction', 'Water Veil', 'Weak Armor', 'White Smoke', 'Wonder Guard', 'Wonder Skin ', 'Zen Mode'] not in index"

Reply:
It looks like its trying to use a bunch of the older columns - try running models.py again (the one in this branch) to make new pickles and give it another go.